src/RF24LogHandler.h

```diff
@@ -0,0 +1,88 @@
+/**
+ * @file RF24LogHandler.h
+ *
+ * Created on: 2 Oct 2020
+ *     Author: Witold Markowski (wmarkow)
+ *
+ * Copyright (C)
+ *    2020        Witold Markowski (wmarkow)
+ *    2021        Brendan Doherty (2bndy5)
+ *
+ * This General Public License does not permit incorporating your program into
+ * proprietary programs.  If your program is a subroutine library, you may
+ * consider it more useful to permit linking proprietary applications with the
+ * library.  If this is what you want to do, use the GNU Lesser General
+ * Public License instead of this License.
+ */
+
+#ifndef SRC_RF24LOGHANDLER_H_
+#define SRC_RF24LOGHANDLER_H_
+
+#include <WString.h>
+#include <stdint.h>
+#include <stdarg.h>
+#include "RF24LogLevel.h"
+
+/**
+ * A base class for handling log messages.
+ */
+class RF24LogHandler
+{
+public:
+      /**
+       * log message.
+       * @param logLevel the level of the logging message
+       * @param vendorId The prefixed origin of the message
+       * @param message The message
+       */
+      void log(uint8_t logLevel,
+                          const __FlashStringHelper *vendorId,
+                          const char *message,
+                          va_list *args);
+
+      /**
+       * log message.
+       * @param logLevel the level of the logging message
+       * @param vendorId The prefixed origin of the message
+       * @param message The message
+       */
+      void log(uint8_t logLevel,
+                          const __FlashStringHelper *vendorId,
+                          const __FlashStringHelper *message,
+                          va_list *args);
+
+      /**
+       * set the maximal level of the logged messages.
+       * @param logLevel maximal level of the logged message
+       */
+      void setLogLevel(uint8_t logLevel);
+
+protected:
+      uint8_t logLevel;
+
+      RF24LogHandler();
+
+      /**
+       * Internal log message.
+       * @param logLevel the level of the logging message
+       * @param vendorId The prefixed origin of the message
+       * @param message The message
+       */
+      virtual void log0(uint8_t logLevel,
+                          const __FlashStringHelper *vendorId,
+                          const char *message,
+                          va_list *args) = 0;
+
+      /**
+       * Internal log message.
+       * @param logLevel the level of the logging message
+       * @param vendorId The prefixed origin of the message
+       * @param message The message
+       */
+      virtual void log0(uint8_t logLevel,
+                          const __FlashStringHelper *vendorId,
+                          const __FlashStringHelper *message,
+                          va_list *args) = 0;

```

Comment:
>Maybe they should be named like write to avoid confusion with log()

yes. that would be more Arduino-like API. I was also confused about the name, but now I see that they're called from the public `log()` Without filtering (or any other manipulation), they should be called `write()`.

Reply:
Done.